MYA first batch (#82)
NHSDigital/prevention-services-design-history: app/posts/manage-your-appointments/2024/10/2024-10-10-designing-availability-view.md

---
title: "Designing the availability and bookings view"
date: 2024-10-10
---

As part of managing availability and appointments, our users need to be able to view availability they’ve created, check how many appointments they have booked, and view a list of daily bookings for their site. 

## What we did 

From our discovery research, we know that users need to: 

> - See how many booked and unbooked appointments they had for each day, to help them understand demand for vaccinations. 
> - See what sessions and services they are providing for each day 

Reviewer: I think these can both be lower-case with no full stop, as there’s a lead-in line?

```suggestion
- see how many booked and unbooked appointments they had for each day, to help them understand demand for vaccinations
- see what sessions and services they are providing for each day
```

### Weekly view

![Screenshot of a page that shows a weekly overivew, with numbers of booked and unbooked appointments](week.png)

The weekly view was where users would find the information they needed to manage their availability, including accessing a list of daily bookings. 

### Findings for iteration 1

Users didn’t understand the purpose of the yearly view, and did not find it useful. 

> It just doesn't have enough flexibility in terms of like I would want to see different numbers. But I don't, it doesn't appear to me that I have any control over that and that can't be.

Users found the weekly overview more useful but were unable to view a breakdown of booked appointments by service or cohort. This is critical information for determining how many vaccines and what types they need to prepare.   

Based on these findings, we concluded that we needed to consider how best to present information in the weekly overview, to ensure that users got the level of detail they needed, and that the yearly overview was not useful in helping our users understand their availability and capacity levels. 

## Iteration 2 

> In V2 of the availability view, we changed the yearly view for a monthly view that would give users a breakdown of appointment totals by week.  We also updated the weekly view, so that the booked appointments were broken down by service or cohort for each day. 

Reviewer: Use 'iteration' instead of 'V2' here for consistency?

```suggestion
In the second iteration of the availability view, we changed the yearly view for a monthly view that would give users a breakdown of appointment totals by week.  We also updated the weekly view, so that the booked appointments were broken down by service or cohort for each day. 
```

### Monthly view

![Screenshot of a page that shows a monthly overview, where users can see appointment data broken down by week](month.png)

### Weekly view

![Screenshot of a page that shows a weekly overivew, with numbers of booked and unbooked appointments](week2.png)

### Daily bookings list

![Screenshot of 2 pages, showing the check your answers pattern, for users to check the details of the sessions they've added](daily-bookings.png)

We introduced a daily bookings list in this iteration, to allow users to see which slots are booked, and who is attending the appointment. 

## Findings for iteration 2
 

> Users found the monthly view more useful, and the breakdown of data was mostly clear.  the term ‘available appointments’ was not understood by all users, and they were unsure how many appointments they had that were not booked. 

Reviewer: ```suggestion
Users found the monthly view more useful, and the breakdown of data was mostly clear.  The term ‘available appointments’ was not understood by all users, and they were unsure how many appointments they had that were not booked. 
```
